lib/telemetry/telemetry.go

```diff
@@ -0,0 +1,81 @@
+package telemetry
+
+import (
+	"time"
+
+	"github.com/armon/go-metrics"
+)
+
+/// TODO(kit) Label stands in for labels
+type Label struct {
+	Key   string
+	Value string
+}
+
+// PLSREVIEW(kit): I decided to directly wrap the API of go-metrics because it informs all of our call sites.
+//  However, this does not need to be set in stone. For example, I renamed Incr to Inc. This is a good opportunity to
+//  adjust the metrics API we use internally. Any suggestions for improvements would be welcome!
+type MetricsClient interface {
```

Comment:
I think this will also need some method similar to the `AddSample` method from armons lib. It appears that we used that in serf.